ZenPacks/zenoss/OpenStackInfrastructure/patches/platform.py

```diff
@@ -57,19 +57,21 @@ def openstackInstance(self):
 
     # Nope?  OK, go to MAC addresses.
     instances = set()
-    macs = getIpInterfaceMacs(self)
     vnics = []
-    for v in catalog_search(
-            self.dmd.Devices,
-            'ZenPacks_zenoss_OpenStackInfrastructure_Vnic',
-            serialNumber=macs):
-        try:
-            vnic = v.getObject()
-        except Exception:
-            # ignore a stale entry
-            pass
-        else:
-            vnics.append(vnic)
+
+    macs = getIpInterfaceMacs(self)
+    if macs:
+        for v in catalog_search(
+                self.dmd.Devices,
+                'ZenPacks_zenoss_OpenStackInfrastructure_Vnic',
+                serialNumber=macs):
+            try:
+                vnic = v.getObject()
+            except Exception:
+                # ignore a stale entry
+                pass
+            else:
+                vnics.append(vnic)
```

Comment:
Okay, so it's the same logic for looking up vnics, bur we're skipping it if there aren't any MACs.  Fair enough.  If there are no MACs, we'll have an empty vnics list, which appears to be what we want.  Looks good to me.